common/src/main/java/com/wynntils/handlers/tooltip/impl/crafted/CraftedTooltipComponent.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.handlers.tooltip.impl.crafted;
+
+import com.wynntils.models.items.properties.CraftedItemProperty;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+
+public abstract class CraftedTooltipComponent<T extends CraftedItemProperty> {
+    public abstract List<Component> buildHeaderTooltip(T craftedItem);
+
+    public abstract List<Component> buildFooterTooltip(T craftedItem);
+
+    protected MutableComponent buildRequirementLine(String requirementName, boolean fulfilled) {
+        MutableComponent requirement;
+
+        requirement = fulfilled
```

Comment:
This checkmark/cross selection from a bool should be a helper function...

Reply:
It's been like that for a bit, I've just copied the method. There are a few refactors the whole tooltip building system could take, I opted to only change what's really needed.